pandas/core/indexes/multi.py

```diff
@@ -243,10 +243,23 @@ def __new__(cls, levels=None, codes=None, sortorder=None, names=None,
 
         if verify_integrity:
             result._verify_integrity()
+
+        codes = [cls._reassign_na_codes(level, code)
+                 for level, code in zip(levels, codes)]
+        result._set_codes(codes, validate=False)
+
         if _set_identity:
             result._reset_identity()
+
         return result
 
+    @classmethod
+    def _reassign_na_codes(cls, level, code):
```

Comment:
call this _validate_codes, pls add a doc-string.

This should be an instance method (and not a classmethod)

Reply:
method level changed and moved into validation method